packages/ui/src/molecules/RadioGroup/RadioGroup.test.tsx

```diff
@@ -0,0 +1,104 @@
+import { render } from '@testing-library/react'
+import { renderHook } from '@testing-library/react-hooks'
+import userEvent from '@testing-library/user-event'
+import type { ReactNode } from 'react'
+import React from 'react'
+
+import { RadioGroup, RadioOption, useRadioOption } from '.'
+
+describe('RadioGroup', () => {
+  it('Should render radio group with radio option', () => {
+    const onChange = jest.fn()
+    const { getByTestId, getAllByLabelText } = render(
+      <RadioGroup name="radio-group" selectedValue="radio" onChange={onChange}>
+        <RadioOption value="radio" label="Radio 1" />
+      </RadioGroup>
+    )
+
+    const [radio] = getAllByLabelText(/Radio 1/i)
+
+    expect(radio).toBeInTheDocument()
+    expect(getByTestId('store-radio-option')).toBeInTheDocument()
+  })
+  it('Should render RadioOption Children', () => {
+    const onChange = jest.fn()
+    const { getByRole } = render(
+      <RadioGroup
+        name="radio-group"
+        selectedValue="radio-1"
+        onChange={onChange}
+      >
+        <RadioOption value="radio-1" label="Radio 1">
+          <h1>Radio Group 1</h1>
+        </RadioOption>
+      </RadioGroup>
+    )
+
+    expect(getByRole('heading')).toHaveTextContent('Radio Group 1')
+  })
+  it('Should pass name from RadioGroup to RadioOptions', () => {
+    const onChange = jest.fn()
+
+    const { container } = render(
+      <RadioGroup
+        name="radio-group"
+        selectedValue="radio-1"
+        onChange={onChange}
+      >
+        <RadioOption value="radio-1" label="Radio 1" />
+        <RadioOption value="radio-1" label="Radio 1" />
```

Comment:
In a real-world scenario, the radio options normally have different `value`s and labels, so I suggest we tweak this example in such way:
```suggestion
        <RadioOption value="radio-1" label="Radio 1" />
        <RadioOption value="radio-2" label="Radio 2" />
```